engine.py

```diff
@@ -177,7 +179,21 @@ def make_move(self, move):
 		self.board[move.end_row][move.end_col] = move.piece_moved
 		self.move_log.append(move) # log move
 		self.light_to_move = not self.light_to_move # next player to move
-
+		#pawnpromotion
+		if move.pawnpromotion:
+			master = tk.Tk()
+			master.title("Pawn Promotion Chooser")
+			master.withdraw()
+			while True:
+				userDb = simpledialog.askstring("pawns promotion chooser",
+												"Enter q to promote to Queen,r to Rook ,b to Bishop or n to Knight: ")
+				n = ["q","n","r","b"]
+				if userDb.lower() not in n:
```

Comment:
if cancel is pressed instead of enter, this line throws an error...can this be fixed?